packages/legend-studio/src/stores/editor-state/element-editor-state/testable/TestableEditorState.ts

```diff
@@ -0,0 +1,203 @@
+    this.testResultState = new TestableTestResultState(this, this.editorStore);
+    this.assertionEditorStates = test.assertions.map(
+      (assertion) =>
+        new TestAssertionEditorState(this.editorStore, assertion, this),
+    );
+    this.selectedAsertionState = this.assertionEditorStates[0];
+    this.isReadOnly = isReadOnly;
+  }
+  setSelectedTab(val: TESTABLE_TEST_TAB): void {
+    this.selectedTab = val;
+  }
+
+  setAssertionToRename(assertion: TestAssertion | undefined): void {
+    this.assertionToRename = assertion;
+  }
+
+  addAssertion(): void {
+    const assertion = createEmptyEqualToJsonAssertion(this.test);
+    atomicTest_addAssertion(this.test, assertion);
+    const assertionState = new TestAssertionEditorState(
+      this.editorStore,
+      assertion,
+      this,
+    );
+    addUniqueEntry(this.assertionEditorStates, assertionState);
+    this.selectedAsertionState = assertionState;
+    this.resetResult();
+  }
+
+  deleteAssertion(assertionState: TestAssertionEditorState): void {
+    deleteEntry(this.test.assertions, assertionState.assertion);
+    deleteEntry(this.assertionEditorStates, assertionState);
+    if (this.selectedAsertionState === assertionState) {
+      this.selectedAsertionState = this.assertionEditorStates[0];
+    }
+  }
+
+  openAssertion(val: TestAssertion): void {
+    const state = this.assertionEditorStates.find((a) => a.assertion === val);
+    if (state) {
+      this.selectedAsertionState = state;
+    }
+  }
+
+  *runTest(): GeneratorFn<void> {
+    try {
+      this.resetResult();
+      this.runningTestAction.inProgress();
+      const input = new RunTestsTestableInput(this.testable);
+      const suite =
+        this.test.__parent instanceof TestSuite
+          ? this.test.__parent
+          : undefined;
+      input.unitTestIds.push(new AtomicTestId(suite, this.test));
+      const testResults =
+        (yield this.editorStore.graphManagerState.graphManager.runTests(
+          [input],
+          this.editorStore.graphManagerState.graph,
+        )) as TestResult[];
+      const result = guaranteeNonNullable(testResults[0]);
+      assertTrue(
+        result.testable === this.testable &&
+          result.atomicTestId.atomicTest === this.test,
+        'Unexpected test result',
+      );
+      this.handleTestResult(result);
+      this.runningTestAction.complete();
+    } catch (error) {
+      assertErrorThrown(error);
+      this.editorStore.applicationStore.notifyError(error);
+      this.runningTestAction.fail();
+    }
+  }
+
+  resetResult(): void {
+    this.testResultState.setResult(undefined);
+    this.assertionEditorStates.forEach((assertionState) =>
+      assertionState.assertionResultState.setTestResult(undefined),
+    );
+  }
+
+  handleTestResult(testResult: TestResult): void {
+    this.testResultState.setResult(testResult);
+    this.assertionEditorStates.forEach((assertionState) => {
+      assertionState.assertionResultState.setTestResult(testResult),
```

Comment:
This looks really weird, did you mean to use a semi-colon instead of a `coma` at the end of this?